src/dymaptic.GeoBlazor.Core/Components/Views/MapView.razor.cs

```diff
@@ -143,87 +145,390 @@ public double Rotation
     [Parameter]
     public bool? AllowDefaultEsriLogin { get; set; }
 
+#region EventHandlers
+
+    /// <summary>
+    ///     Surfaces JavaScript errors to the .NET Code for debugging.
+    /// </summary>
+    /// <param name="message">
+    ///     The JavaScript error message.
+    /// </param>
+    /// <param name="name">
+    ///     The name of the JavaScript error.
+    /// </param>
+    /// <param name="stack">
+    ///     The JavaScript stack trace.
+    /// </param>
+    /// <exception cref="JavascriptException">
+    ///     Wraps the JS Error and throws a .NET Exception.
+    /// </exception>
+    [JSInvokable]
+    public void OnJavascriptError(string message, string name = "", string stack = "")
+    {
+#if DEBUG
+        ErrorMessage = message.Replace("\n", "<br>");
+        StateHasChanged();
+#endif
+        var exception = new JavascriptException(message, name, stack);
+
+        if (OnJavascriptErrorHandler is not null)
+        {
+            OnJavascriptErrorHandler?.Invoke(exception);
+        }
+        else
+        {
+            throw exception;
+        }
+    }
+    
     /// <summary>
-    ///     Handler delegate for click actions on the view. Must take in a <see cref="Point"/> and return a <see cref="Task"/>.
+    ///     Implement this handler in your calling code to catch and handle Javascript errors.
     /// </summary>
     [Parameter]
+    public Func<JavascriptException, Task>? OnJavascriptErrorHandler { get; set; } 
+
+    /// <summary>
+    ///     JS-Invokable method to return view clicks.
+    /// </summary>
+    /// <param name="mapPoint">
+    ///     The <see cref="Point"/> that was clicked.
+    /// </param>
+    [JSInvokable]
+    public async Task OnJavascriptClick(Point mapPoint)
+    {
+#pragma warning disable CS0618
+        OnClickAsyncHandler?.Invoke(mapPoint);
+#pragma warning restore CS0618
+        await OnClick.InvokeAsync(mapPoint);
+    }
+    
+    /// <summary>
+    ///     Handler delegate for click events on the view. Must take in a <see cref="Point"/> and return a <see cref="Task"/>.
+    /// </summary>
+    /// <remarks>
+    ///     <b style="color: red">OBSOLETE: Use OnClick EventCallback instead</b>
+    /// </remarks>
+    [Parameter]
+    [Obsolete("Use OnClick EventCallback instead.")]
     public Func<Point, Task>? OnClickAsyncHandler { get; set; }
+    
+    /// <summary>
+    ///     Handler delegate for click events on the view.
+    /// </summary>
+    [Parameter]
+    public EventCallback<Point> OnClick { get; set; }
```

Comment:
In general, I think these events should return the same data that the ArcGIS even returns. For example, HitTest returns a full mouse event with additional properties attached. This way you can do things like check for modifier keys being pressed, etc.

Reply:
Agreed, I will go back through and update the returns to all be full event objects.